src/nvim/message.c

```diff
@@ -3428,30 +3420,37 @@ int vim_vsnprintf(char *str, size_t str_m, char *fmt, va_list ap, typval_T *tvs)
           f[f_l++] = fmt_spec;
           f[f_l++] = '\0';
 
-          if (fmt_spec == 'p')
-            str_arg_l += sprintf(tmp + str_arg_l, f, ptr_arg);
-          else if (fmt_spec == 'd') {
+          if (fmt_spec == 'p') {
+            str_arg_l += (size_t)sprintf(tmp + str_arg_l, f, ptr_arg);
```

Comment:
Handling of negative values was my concern. But I think the result is the same with or without the `size_t` cast, no matter what the return value is.